src/query/query_executor.h

```diff
@@ -967,6 +967,7 @@ struct xasl_node
   const char *query_alias;
   int dbval_cnt;		/* number of host variables in this XASL */
   bool iscan_oid_order;
+  int max_iterations;
```

Comment:
There should be no issue since the parameter is read at execution.
I added code to reset xasl->max_iterations to -1.